### ISSUE TITLE
Https not supported

### ISSUE THREAD

**Issue by lukehansford** [koying/SPMC#824]
It seems that https is not supported by kodi 16.

Many stream providers are starting to move over to HTTPS now which Krypton supports.

This means by using SPMC we are missing out on sources.

Is it possible to add support for HTTPS to SPMC or will SPMC be moving over to the Kodi 17 codebase soon?

I wish to use SPMC for voice search, some builds still on 16 like pulse and on android but the lack of HTTPS could mean alot of people moving to Krypton.

**Comment by lukehansford** [comment]
Can this be added to SPMC or must we wait for kodi 17

**Comment by Strokerace1** [comment]
First off, what do you mean that https doesn't work? I have used links with https in 15. All https means is its a secured and encrypted. Its been like that since 2015. So there is no reason for them not to work in SPMC.

**Comment by lukehansford** [comment]
Kodi 16.1 does not support the use of SSL https.

This explains the issue

http://koditips.com/kodi-exodus-not-working-click-here-for-info/

**Comment by Strokerace1** [comment]
Yes it does. It has supported it since 2015. Maybe you should do your research. All this is stuff from Kodi is to get people to upgrade their boxes. Which is fine for Kodi's death. No body is using exodus any more as its poorly made. The crap about TLSv1 and TLSv2 is just smoke and mirrors. As it stands, there is no TLSv2, its TLSv1.2 and SSL2. And the fact about python not handling is BS. There is enough documentation on the net to show otherwise and how false that statement is. Further more, I am sure that they would have update python in kodi back in 2014 when python 2.7.9 came out. As this is part of the Android O/S and not really part of Kodi. 

https from Kodi 
http://kodi.wiki/view/HTTP

**Comment by ghost** [comment]
I believe the issue is to do with the version of python that's bundled in the Android platform of Kodi - it's much older compared to the others. Python 2.7.9 introduced TLS v1.1/1.2 support.

The problem arising with stream providers is down to many hosting themselves behind CloudFlare which use a strict cipher-suite in their HTTP configurations: only TLS v1.1 / v1.2 and experimental v1.3 due to the recent POODLE vunlerabilty in SSL and TLS v1.0.

**Comment by sbaghdadi** [comment]
There is already a ticket for that issue: https://github.com/koying/SPMC/issues/568#issuecomment-268343934
